Purchases/Public/RCPurchases.h

```diff
@@ -415,11 +415,18 @@ NS_SWIFT_NAME(purchases(_:didReceiveUpdated:));
 
 /**
  Called when a user initiates a promotional in-app purchase from the App Store. If your app is able to handle a purchase at the current time, run the deferment block in this method. If the app is not in a state to make a purchase: cache the defermentBlock, then call the defermentBlock when the app is ready to make the promotional purchase. If the purchase should never be made, you don't need to ever call the defermentBlock and `RCPurchases` will not proceed with promotional purchases.
- 
+
  @param product `SKProduct` the product that was selected from the app store
  */
 - (void)purchases:(RCPurchases *)purchases shouldPurchasePromoProduct:(SKProduct *)product defermentBlock:(RCDeferredPromotionalPurchaseBlock)makeDeferredPurchase;
 
+
+/**
+ Updates the purchaser information.
+ This method should only be called when a purchase has been made through the RevenueCat server, but not through the app, to force a refresh.
+ */
+- (void)updatePurchaserInfoCache;
```

Comment:
@vegaro thoughts on naming for the method? I didn't want to call it `invalidateCache` since it does more than that. And I also didn't want to just go with `updateCache` since it's already in use by the class. Maybe that could be refactored.

Reply:
also, I went with the version that would also fire the delegate method, so that the clients know when to refresh.

Reply:
How can we refactor this? We could just expose `- (void)updateCachesWithCompletionBlock:(nullable RCReceivePurchaserInfoBlock)completion`, but then people would also be refreshing offerings, and we might not want that extra call... 

We could also expose `- (void)updatePurchaserInfoCache:(nullable RCReceivePurchaserInfoBlock)completion`, but then we would need to call `[self.deviceCache resetCachesTimestamp]` . somewhere. We should have independent timestamps for both offerings and purchaser infos.

What if we rename `- (void)updatePurchaserInfoCache:(nullable RCReceivePurchaserInfoBlock)completion` to `fetchAndCachePurchaserInfo` (same for the offerings one).

Reply:
for anyone reading - we discussed in person. The takeaway was: 
- The issue is that this would also force a refresh on offerings, which is not necessary and is an extra backend call. And every time you call the backend unnecessarily, a panda dies. 
- The offerings get refreshed because `RCDeviceCache` stores only one date for last update, which is shared for purchaser info and offerings. 
- We're going to separate that into two dates (actually, a wrapper object of which `RCDeviceCache` will have two instances, one for purchaser info and one for offerings). Then, only one of those will be refreshed from this method.

Reply:
I just realized in Android the names are `fetchAndCache...` https://github.com/RevenueCat/purchases-android/blob/master/purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt#L670

Reply:
Honestly, I feel like `updatePurchaserInfoCache` and `fetchAndCache` are both internal methods and exposing them directly might be a mistake, especially if we refactor something internal and forget that they are exposed.

A new method that just invalidates the cache, doesn't a trigger a refresh might produce the most low touch result. i.e.
`invalidatePurchaserInfoCache`

We shouldn't really worry about what methods we have right now in the internal implementation.

Reply:
I agree having `invalidatePurchaserInfoCache` would probably be better since it's more atomic and  does less things. That was actually my initial idea.

Reply:
sure. My thinking with having this also force a refresh was that as a developer, there's a good way that you have stuff hooked up to the updated purchaser info event, so the way that this would look is: 
- you get a silent push from your server saying that you granted promotional from somewhere else
- you call re-fetch
- whatever hooks you already had in place are automatically called when purchaser info updates, no need to do anything

whereas if you only invalidate cache, you have an extra step of: 
- you get a silent push from your server saying that you granted promotional from somewhere else
- you call invalidate cache
- you manually force a re-fetch by asking for the purchaser info after invalidating cache. 
- whatever hooks you already had in place are automatically called when purchaser info updates

like, I don't see a scenario where you're calling invalidate cache and you don't actually want to refresh right away. 
I see the concern about `updatePurchaserInfoCache` being exposed, but I don't feel like it's a big distance from there to asking devs to `invalidatePurchaserInfoCache` + check purchaser info when done.

Reply:
I'll make the update